testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -889,6 +889,112 @@ public void testTrinoSparkConcurrentInsert()
+                " USING %s";
+        onSpark().executeQuery(format(sparkTableDefinition, defaultCatalogTableName, storageFormat.name()));
+
+        String insert = "" +
+                "INSERT INTO TABLE %s SELECT" +
+                "  'Doc213'" +
+                ", map('s1', array(named_struct('sname', 'ASName1', 'snumber', 201), named_struct('sname', 'ASName2', 'snumber', 202)))" +
+                ", array(map('m1', array(named_struct('mname', 'MAS1Name1', 'mnumber', 301), named_struct('mname', 'MAS1Name2', 'mnumber', 302)))" +
+                "       ,map('m2', array(named_struct('mname', 'MAS2Name1', 'mnumber', 401), named_struct('mname', 'MAS2Name2', 'mnumber', 402))))" +
+                ", named_struct('id', 1, 'name', 'P. Sherman', 'address', named_struct('street_number', 42, 'street_name', 'Wallaby Way'))";
+        onSpark().executeQuery(format(insert, defaultCatalogTableName));
+        onSpark().executeQuery(format("CALL system.migrate('%s')", defaultCatalogTableName));
+
+        String sparkTableName = sparkTableName(baseTableName);
+        Row row = row("Doc213", "ASName2", 201, "MAS2Name1", 302, "P. Sherman", 42, "Wallaby Way");
+
+        String sparkSelect = "SELECT" +
+                "  doc_id" +
+                ", nested_map['s1'][1].sname" +
+                ", nested_map['s1'][0].snumber" +
+                ", nested_array[1]['m2'][0].mname" +
+                ", nested_array[0]['m1'][1].mnumber" +
+                ", nested_struct.name" +
+                ", nested_struct.address.street_number" +
+                ", nested_struct.address.street_name" +
+                "  FROM ";
+
+        QueryResult sparkResult = onSpark().executeQuery(sparkSelect + sparkTableName);
+        // The Spark behavior when the default name mapping does not exist is not consistent
+        assertThat(sparkResult).containsOnly(row);
+
+        String trinoSelect = "SELECT" +
+                "  doc_id" +
+                ", nested_map['s1'][2].sname" +
+                ", nested_map['s1'][1].snumber" +
+                ", nested_array[2]['m2'][1].mname" +
+                ", nested_array[1]['m1'][2].mnumber" +
+                ", nested_struct.name" +
+                ", nested_struct.address.street_number" +
+                ", nested_struct.address.street_name" +
+                "  FROM ";
+
+        String trinoTableName = trinoTableName(baseTableName);
+        QueryResult trinoResult = onTrino().executeQuery(trinoSelect + trinoTableName);
+        assertThat(trinoResult).containsOnly(row);
+
+        // After removing the name mapping, columns from migrated files should be null since they are missing the Iceberg Field IDs
+        onSpark().executeQuery(format("ALTER TABLE %s UNSET TBLPROPERTIES ('schema.name-mapping.default')", sparkTableName));
+        assertThat(onTrino().executeQuery(trinoSelect + trinoTableName)).containsOnly(row(null, null, null, null, null, null, null, null));
```

Comment:
thanks for these tests, they're great! do we know what is the spark behavior when property is missing?

Reply:
ah, https://github.com/trinodb/trino/issues/9843#issuecomment-970474590